clap_complete/src/dynamic/completer.rs

```diff
@@ -704,3 +723,110 @@ impl CompletionCandidate {
         self.visible
     }
 }
+
+/// User-provided completion candidates for an argument.
+///
+/// This is useful when predefined value hints are not enough.
+pub trait CustomCompleter: core::fmt::Debug + Send + Sync {
+    /// All potential candidates for an argument.
+    ///
+    /// See [`CompletionCandidate`] for more information.
+    fn completions(&self) -> Vec<CompletionCandidate>;
+}
+
+/// Implement `CustomCompleter` for a closure
+///
+/// # Example
+///
+/// ```rust
+/// use clap_complete::dynamic::{CustomCompleter, CompletionCandidate};
+///
+/// let completer = || {
+///     let content = std::fs::read_to_string("/path/to/file");
+///     match content {
+///         Ok(content) => {
+///             content.lines().map(|os| CompletionCandidate::new(os).visible(true)).collect()
+///         }
+///         Err(_) => vec![],
+///     }
+/// };
+///
+/// fn main() {
+///     let completer = ArgValueCompleter::new(MyCustomCompleter{
+///         file: std::path::PathBuf::from("/path/to/file"),
+///     });
```

Comment:
There are two different `completer`s in here, I don't think we are getting the one we intended